jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerManager.kt

```diff
@@ -616,12 +626,60 @@ class CodeModernizerManager(private val project: Project) : PersistentStateCompo
                 CodetransformTelemetry.totalRunTime(
                     codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId(),
                     codeTransformResultStatusMessage = "JobCancelled",
-                    codeTransformRunTimeLatency = calculateTotalLatency(CodeTransformTelemetryState.instance.getStartTime(), Instant.now())
+                    codeTransformRunTimeLatency = calculateTotalLatency(CodeTransformTelemetryState.instance.getStartTime(), Instant.now()),
+                    codeTransformLocalJavaVersion = getJavaVersionFromProjectSetting(),
+                    codeTransformLocalMavenVersion = getMavenVersions(),
                 )
             }
         }
     }
 
+    private fun getJavaVersionFromProjectSetting(): String? = project.tryGetJdk()?.toString()
+
+    private fun getMavenVersions(): String {
+        fun getVersion(mavenCommand: String): String? {
+            try {
+                val commandLine = GeneralCommandLine(listOf(mavenCommand, "-v"))
+                    .withWorkDirectory(project.basePath)
+                    .withRedirectErrorStream(true)
+                val output = ExecUtil.execAndGetOutput(commandLine)
+                if (output.exitCode == 0) {
+                    return parseMavenVersion(output.stdout)
+                } else {
+                    LOG.error { "Failed to fetch $mavenCommand version: ${output.stdout}" }
+                }
+            } catch (e: ProcessNotCreatedException) {
+                LOG.warn { "$mavenCommand not set up" }
+            } catch (e: Exception) {
+                LOG.error(e) { "Failed to fetch $mavenCommand version" }
+            }
+            return null
+        }
+
+        // Get local maven version
+        val localMavenVersion: String? = getVersion("mvn")
+
+        // Get wrapper maven version
+        val mvnw = if (SystemInfo.isWindows) "./mvnw.cmd" else "./mvnw"
+        val wrapperMavenVersion: String? = getVersion(mvnw)
+
+        // Get user's Maven setting (using bundled vs local vs wrapper)
+        val mavenSettings = MavenProjectsManager.getInstance(project).getGeneralSettings()
+        val mavenHome = mavenSettings.getMavenHome()
+        // TODO: Need to detect bundled Maven version that come with IDEA
+        //  The utility returns "Use Maven wrapper" if using wrapper, "Bundled (Maven 3)" if using Bundled Maven, otherwise the local maven version.
+        val userMavenSetting = MavenUtil.getMavenVersion(mavenHome) ?: mavenHome
+
+        return "$wrapperMavenVersion (mvnw) -- $localMavenVersion (mvn) -- user setting: $userMavenSetting"
+    }
+
+    private fun parseMavenVersion(output: String?): String? {
```

Comment:
Question: Should this function be in a utility class? The manager file shouldn't be bundled up with utility functions.

Reply:
Good call, can move it to util.